pages/studentPortal/web/index.tsx

```diff
@@ -1,16 +1,18 @@
 import { useMutation, useQuery } from "@apollo/client";
 import moment from "moment";
 import React, { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
 import UnitView from "../../../components/studentPortal/lessons/UnitView";
 import PageHeader from "../../../components/ui/PageHeader";
 import { FETCH_USER_INTRO_NODES } from "../../../graphql/studentPortal/courses/fetchUserIntroNodes";
 import { UPDATE_USER } from "../../../graphql/studentPortal/users/updateUser";
 import { useAuth } from "../../../lib/authContext";
+import { profileSelector } from "../../../redux/profileSlice";
 import { reactUnits, Unit } from "../../api/studentPortal/units";
 
 export default function StudentPortalPage() {
   const { user } = useAuth();
-
+  const { userRole } = useSelector(profileSelector);
```

Comment:
It's fine for the page to need to know about the user role, but our lower level components should not.

Reply:
The page is the higher level orchestrator that connects everything.

Reply:
removed userRole from other components.  Created freemium components instead and checked userRole on the index page and deployed components accordingly from there.